instrumentation_test.go

```diff
@@ -16,6 +16,9 @@ func collectLogEvent(t *testing.T, f func()) map[string]interface{} {
 	r, w := io.Pipe()
 	defer r.Close()
 	log.SetOutput(w)
+	t.Cleanup(func() {
+		log.SetOutput(log.New().Out)
+	})
```

Comment:
Instead of assuming `log.New().Out` is the correct file I would save the previous value of `Out` in case other redirections have been done.
```golang
prevOut := log.Out
log.SetOutput(w)
t.Cleanup(func() {
  log.SetOutput(prevOut)
})
```